### DISCUSSION TITLE
Create a set of IED files with varying structure parameters

### DISCUSSION THREAD

**Discussion by joe-pierce** [People-Places-Solutions/floodmodeller-api#8] [Show and tell]
## Creating IED files with varying parameters

Here's a simple script that takes the conduit structure from a DAT file (in this case called 'XS06_06Cin2') and creates a set of IED files each representing a different conduit diameter. 
This could be useful for sensitivity testing of a structure.

The code:

``` python
from pathlib import Path
from floodmodeller_api import IED, DAT

dat_file = Path("data/A1_Northumberland_v08.DAT")

dat = DAT(dat_file)
culvert = dat.conduits['XS06_06Cin2']

for diameter in [0.5, 0.6, 0.7, 0.8]:
    ied = IED()
    culvert.diameter = diameter
    ied.conduits[culvert.name] = culvert
    ied.save(Path(dat_file.parent, f'{dat_file.stem}_{int(diameter*1000)}mm_diameter.IED'))
```

Produces:
![image](https://user-images.githubusercontent.com/56606086/167425109-e0caa958-a327-4f4f-bee3-7b44ceff3bb4.png)

**Comment by Amrie271** [comment]
Can this be set up for changes in return periods, also a model that is linked to 2D tuflow?

**Reply by joe-pierce** [reply]
Hi - Yes you could reference different IED files in your IEF based on return period if that's what you mean? The exact logic of the code would depend on the structure and naming of your files but should all be do-able :smile: 
Also yes any model linked to Tuflow would still link as normal through the IEF.
